lib/topological_inventory/providers/common/operations/sources_api_client.rb

```diff
@@ -20,21 +22,23 @@ def init_default_api
 
           def fetch_default_endpoint(source_id)
             endpoints = api.list_source_endpoints(source_id)&.data || []
-            endpoint = endpoints.find(&:default)
-
-            raise "Sources API: Endpoint not found! (source id: #{source_id})" if endpoint.nil?
-
-            endpoint
+            endpoints.find(&:default)
           end
 
-          def fetch_authentication(source_id, default_endpoint = nil)
+          def fetch_authentication(source_id, default_endpoint = nil, authtype = nil)
             endpoint = default_endpoint || fetch_default_endpoint(source_id)
             return if endpoint.nil?
 
             endpoint_authentications = api.list_endpoint_authentications(endpoint.id.to_s).data || []
             return if endpoint_authentications.empty?
 
-            auth_id = endpoint_authentications.first.id
+            auth_id = if authtype.nil?
+                        endpoint_authentications.first&.id
+                      else
+                        endpoint_authentications.detect { |a| a.authtype = authtype }&.id
+                      end
+            return if auth_id.nil?
```

Comment:
So - I don't see this being used yet. Is this mainly so it'll only use the same auth type when grabbing the auth? Is this mainly to get around the potentially many authentications issue?

Reply:
@lindgrenj6 authtype is used in Amazon and Azure repos:
- https://github.com/RedHatInsights/topological_inventory-amazon/pull/66/files#diff-03ea31ba8ba5089fdaa66797d8209053L49
- https://github.com/RedHatInsights/topological_inventory-azure/pull/31/files#diff-43c857599ea60dfd8011c249f89703baL49